vowpalwabbit/binary.cc

```diff
@@ -4,44 +4,62 @@
 
 #include <cfloat>
 #include "reductions.h"
+#include "debug_log.h"
 
 using namespace VW::config;
+using std::endl;
 
-template <bool is_learn>
-void predict_or_learn(char&, VW::LEARNER::single_learner& base, example& ec)
-{
-  if (is_learn)
-    base.learn(ec);
-  else
-    base.predict(ec);
+VW_DEBUG_ENABLE(false)
```

Comment:
could this be set thru a compiler flag via cmake? or it could also become a first class citizen like audit

Reply:
This macro defines a const static int that is used to turn on/off debug and therefore compile out debug statements in individual files.  I don't see how we can use a compiler flag unless we define a per file compile time constant.  That seems excessivly complicated.  This is the same mechanism introduced in the predict/learn semantics PR to debug chains of reductions at different levels.